common/authenticator-requests/src/v5/registration.rs

```diff
@@ -143,24 +143,22 @@ impl GatewayClient {
         private_ips: IpPair,
         nonce: u64,
     ) -> Self {
-        // convert from 1.0 x25519-dalek private key into 2.0 x25519-dalek
-        #[allow(clippy::expect_used)]
-        let static_secret = x25519_dalek::StaticSecret::from(local_secret.to_bytes());
-        let local_public: x25519_dalek::PublicKey = (&static_secret).into();
+        let local_public = PublicKey::from(local_secret);
+        let remote_public = PublicKey::from(remote_public);
 
-        let dh = static_secret.diffie_hellman(&remote_public);
+        let dh = local_secret.diffie_hellman(&remote_public);
 
         // TODO: change that to use our nym_crypto::hmac module instead
         #[allow(clippy::expect_used)]
-        let mut mac = HmacSha256::new_from_slice(dh.as_bytes())
+        let mut mac = HmacSha256::new_from_slice(&dh[..])
             .expect("x25519 shared secret is always 32 bytes long");
 
         mac.update(local_public.as_bytes());
         mac.update(private_ips.to_string().as_bytes());
         mac.update(&nonce.to_le_bytes());
 
         GatewayClient {
-            pub_key: PeerPublicKey::new(local_public),
+            pub_key: PeerPublicKey::new(local_public.into()),
```

Comment:
I think that `into()` might be reduntant? though if clippy is happy, I'm happy

Reply:
The `into()` is not redundant, again this is the difference between `nym_crypto::asymmetric::encryption::PublicKey` and  `nym_wireguard_types::PeerPublicKey` which don't have direct conversion, though they both wrap `x25519_dalek` things. so the conversion here is `nym_crypto -> x25519_dalek -> nym_wireguard_types` .